bpf/bpf_lxc.c

```diff
@@ -554,6 +554,24 @@ static inline int handle_ipv4_from_lxc(struct __sk_buff *skb, __u32 *dstID)
 			ep_tail_call(skb, CILIUM_CALL_IPV4_NODEPORT_REVNAT);
 			return DROP_MISSED_TAIL_CALL;
 		}
+# ifdef ENABLE_DSR
+		if (ct_state.dsr) {
```

Comment:
Do we have a case for DSR outside of nodeport? Maybe this should be moved under `ct_state.node_port` instead to avoid two extra tests on bitfields in fast-path when not hit?

Reply:
Discussed offline: we might support enabling of DSR for some not all services (via k8s annotations), so keeping as it is for now.